### DISCUSSION TITLE
Accessing iModels via clientId and client secret

### DISCUSSION THREAD

**Discussion by hvitek** [iTwin/community#33] [Q&A]
Can someone help me or point me in the right direction?

On https://developer.bentley.com/, in the myapps section, I created two applications, the first generated as a WebApp, the second generated as a Service.

I'm implementing in itwin.js environment for viewing itwin models

If I log in to the first one through a pop-up window with a user email and password, everything works as it should.

If I log in to the second one through clientid and client secret, the application logs in, the access token is valid, but I get an error "Requested iModel is not available."
![image](https://github.com/iTwin/community/assets/24455656/b0192d21-16a4-4487-9ee0-79ec63e8b07c)

**Comment by aruniverse** [comment]
Make sure you have added your client credential service account as a member of the iTwin you are trying to access.

The account will look like `service-{XYZ}@apps.imsoidc.bentley.com`
